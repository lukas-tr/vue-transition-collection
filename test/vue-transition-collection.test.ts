import * as Vue from "vue"
import VueTransitionCollection from "../src/vue-transition-collection"
import { mount, createLocalVue } from "@vue/test-utils"

const localVue = createLocalVue()

let components: any = {}

beforeAll(() => {
  components = VueTransitionCollection.generateComponents(Vue as any)
})

const testTransition = (transitionName: string) => {
  it("exists", () => {
    mount(components[transitionName], { localVue })
  })

  it("matches snapshot", () => {
    const wrapper = mount(components[transitionName], {
      localVue,
      slots: {
        default: ["<div>text</div>"]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("uses props", () => {
    const wrapper = mount(components[transitionName], {
      localVue,
      propsData: {
        inverse: true,
        direction: "toggle",
        group: true
      },
      slots: {
        default: ["<div>text</div>"]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
}

describe("Transitions", () => {
  ;["fade", "zoom", "bounce", "flip", "fadeBig", "rotate", "slide"].forEach(componentName => {
    describe(`Testing "${componentName}" transition`, () => {
      testTransition(componentName)
    })
  })
})

describe("Plugin", () => {
  test("install()", () => {
    localVue.use(VueTransitionCollection)
  })

  test("applyDurationAndDelay()", () => {
    const el = document.createElement("div")
    VueTransitionCollection.applyDurationAndDelay({ duration: 100, delay: 200 })(el)
    expect(el.style).toMatchObject({
      animationDuration: "100ms",
      animationDelay: "200ms"
    })
  })
})
