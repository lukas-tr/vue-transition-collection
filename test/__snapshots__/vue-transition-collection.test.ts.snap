// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transitions Testing "bounce" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "bounce" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "fade" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "fade" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "fadeBig" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "fadeBig" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "flip" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "flip" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "rotate" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "rotate" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "slide" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "slide" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;

exports[`Transitions Testing "zoom" transition matches snapshot 1`] = `"<template></template>"`;

exports[`Transitions Testing "zoom" transition uses props 1`] = `"<template inverse=\\"true\\" direction=\\"toggle\\" group=\\"true\\"></template>"`;
